src/argus_htmx/dateformat/views.py

```diff
@@ -21,10 +21,10 @@ def dateformat_names(request: HtmxHttpRequest) -> HttpResponse:
 
 @require_POST
 def change_dateformat(request: HtmxHttpRequest) -> HttpResponse:
-    datetime_format_name = request.POST.get("dateformat", DATETIME_DEFAULT)
-    default_format = DATETIME_FORMATS[DATETIME_DEFAULT]
-    datetime_format = DATETIME_FORMATS.get(datetime_format_name, default_format)
-    request.session["datetime_format"] = datetime_format
-    request.session["datetime_format_name"] = datetime_format_name
-    messages.success(request, f'Switched dateformat to "{datetime_format_name}"')
+    prefs = request.user.get_preferences("argus_htmx")
```

Comment:
Maybe we can prefetch (select_related) all a user's preferences?

Reply:
Where? When?

Reply:
when fetching the request's user. I was thinking it saves us some db round trips ,but perhaps it's not the effort. We currently have one or two queries for preferences, one in the context processor, and perhaps one in the view. So any gains are probably minor

Reply:
request.user is fetched in the auth middleware IIRC.

Reply:
right, that would probably make it too tricky, for only a minor gain